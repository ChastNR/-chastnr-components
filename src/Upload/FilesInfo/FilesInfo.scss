.upl__info {
  word-wrap: break-word;
  color: #959ba4;
  display: block;
  font-size: 13px;
  line-height: 16px;
  margin-top: 4px;
}

.upl__fls {
  max-width: 450px;
  width: 100%;
  margin: 0;
  padding: 0;
  list-style-type: none;
}
