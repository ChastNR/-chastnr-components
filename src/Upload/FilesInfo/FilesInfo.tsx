import { UploadedFile } from '../UploadedFile';

import './FilesInfo.scss';

interface IFilesInfoProps {
  accept?: string;
  info?: React.ReactNode;
  files: File[];
  onDeleteClick: (fileName: string) => void;
}

export const FilesInfo: React.FC<IFilesInfoProps> = ({ accept, files, info, onDeleteClick }) => (
  <div>
    <ul className="upl__fls">
      {files.map((file) => (
        <UploadedFile accept={accept} file={file} key={file.name} onDeleteClick={onDeleteClick} />
      ))}
    </ul>
    <div className="upl__info">{info}</div>
  </div>
);
