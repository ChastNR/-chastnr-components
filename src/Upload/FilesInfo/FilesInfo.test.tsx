import { render } from '@testing-library/react';
import { FilesInfo } from './FilesInfo';

describe('FilesInfo', () => {
  const testFileName = 'test.txt';
  const testFile = new File(['test'], testFileName, {
    type: 'text/plain',
  });

  const mockOnDeleteClick = jest.fn();

  it('should render properly', () => {
    const filesInfo = render(
      <FilesInfo
        accept=".txt"
        files={[testFile]}
        info="testInfo"
        onDeleteClick={mockOnDeleteClick}
      />
    );

    expect(filesInfo.container.firstChild).toMatchSnapshot();
  });
});
