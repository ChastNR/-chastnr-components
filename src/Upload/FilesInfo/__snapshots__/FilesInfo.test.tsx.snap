// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilesInfo should render properly 1`] = `
<div>
  <ul
    class="upl__fls"
  >
    <li
      class="upl__fl"
    >
      <div
        class="upl__fl__cntr"
      >
        <div
          class="upl__fl__cntr__vldt valid"
        >
          <span
            class="rnd__tick"
          >
            <span>
              content
            </span>
          </span>
        </div>
        <div
          class="upl__fl__cntr__data"
        >
          <span
            class="upl__fl__cntr__name"
          >
            test
          </span>
          <span>
            .
            txt
          </span>
          <span
            class="upl__fl__cntr__size"
          >
            0.00 KB
          </span>
        </div>
        <button
          class="upl__fl__cntr__dlt__btn"
          type="button"
        >
          <span
            class="close"
          >
            <span>
              content
            </span>
          </span>
        </button>
      </div>
    </li>
  </ul>
  <div
    class="upl__info"
  >
    testInfo
  </div>
</div>
`;
